proto-public/pbmesh/v1alpha1/proxy.proto

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+syntax = "proto3";
+
+package hashicorp.consul.mesh.v1alpha1;
+
+import "google/protobuf/struct.proto";
+import "pbcatalog/v1alpha1/selector.proto";
+import "pbmesh/v1alpha1/connection.proto";
+import "pbmesh/v1alpha1/expose.proto";
+import "pbmesh/v1alpha1/routing.proto";
+
+message ProxyConfiguration {
+  // Selection of workloads this proxy configuration should apply to.
+  // These can be prefixes or specific workload names.
+  hashicorp.consul.catalog.v1alpha1.WorkloadSelector workloads = 1;
+
+  // dynamic_config is the configuration that could be changed
+  // dynamically (i.e. without needing restart).
+  DynamicConfig dynamic_config = 2;
+
+  // bootstrap_config is the configuration that requires proxies
+  // to be restarted to be applied.
+  BootstrapConfig bootstrap_config = 3;
+
+  // deprecated: prevent usage when using v2 APIs directly.
+  // needed for backwards compatibility
+  google.protobuf.Struct opaque_config = 4 [deprecated = true];
```

Comment:
Should we use `[deprecated = true]` for workload's deprecated fields too (e.g. `connect_native`)?

Reply:
Yes, I think so.